src/snowflake/connector/cursor.py

```diff
@@ -1109,9 +1113,10 @@ def reset(self):
 
     def __iter__(self):
         """Iteration over the result set."""
+        # set _result if _result_set is not None
         if self._result is None and self._result_set is not None:
             self._result = iter(self._result_set)
-        return iter(self._result)
+        return self._result_iterator()
```

Comment:
After setting `self._result` it returns the iterator `self._result_iterator()` which now supports raising exceptions.